contracts/DocumentStore.sol

```diff
@@ -1,103 +1,80 @@
 pragma solidity ^0.5.10;
 
-import "./Ownable.sol";
-
-contract DocumentStore is Ownable {
-    string public name;
-    string public version = "2.3.0";
-
-    /// A mapping of the document hash to the block number that was issued
-    mapping(bytes32 => uint256) documentIssued;
-    /// A mapping of the hash of the claim being revoked to the revocation block number
-    mapping(bytes32 => uint256) documentRevoked;
-
-    event DocumentIssued(bytes32 indexed document);
-    event DocumentRevoked(bytes32 indexed document);
-
-    constructor(string memory _name) public {
-        name = _name;
-    }
-
-    function issue(bytes32 document) public onlyOwner onlyNotIssued(document) {
-        documentIssued[document] = block.number;
-        emit DocumentIssued(document);
-    }
-
-    function bulkIssue(bytes32[] memory documents) public {
-        for (uint256 i = 0; i < documents.length; i++) {
-            issue(documents[i]);
-        }
-    }
-
-    function getIssuedBlock(bytes32 document)
-        public
-        view
-        onlyIssued(document)
-        returns (uint256)
-    {
-        return documentIssued[document];
-    }
-
-    function isIssued(bytes32 document) public view returns (bool) {
-        return (documentIssued[document] != 0);
-    }
-
-    function isIssuedBefore(bytes32 document, uint256 blockNumber)
-        public
-        view
-        returns (bool)
-    {
-        return
-            documentIssued[document] != 0 && documentIssued[document] <= blockNumber;
+import "@openzeppelin/upgrades/contracts/Initializable.sol";
+import "@openzeppelin/contracts-ethereum-package/contracts/ownership/Ownable.sol";
+
+contract DocumentStore is Initializable, Ownable {
```

Comment:
hmmmmmmmmm........ we do lose the ability to interact with proxied contracts on etherscan

Reply:
yeah that's the main drawback he told us, is it that important ?

Reply:
Yep apparently RJ shared that if it's a minimal proxy the etherscan interaction is copied over, but admin upgradability contract isnt. Will not be a problem once we build an api over this.